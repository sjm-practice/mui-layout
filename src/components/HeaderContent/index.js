import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import { isWidthUp } from "@material-ui/core/withWidth";
import InputBase from "@material-ui/core/InputBase";
import IconButton from "@material-ui/core/IconButton";
import SearchIcon from "@material-ui/icons/Search";
import MoreVertIcon from "@material-ui/icons/MoreVert";
import NotificationsIcon from "@material-ui/icons/Notifications";
import PowerSettingsNewIcon from "@material-ui/icons/PowerSettingsNew";
import Typography from "@material-ui/core/Typography";

const styles = ({ spacing, transitions, breakpoints, palette, shape }) => ({
  header: {
    fontWeight: 900,
    minWidth: 0,
    fontSize: 18,
  },
  grow: {
    flexGrow: 1,
  },
  search: {
    position: "relative",
    marginRight: 8,
    borderRadius: shape.borderRadius,
    background: palette.grey[200],
    "&:hover": {
      background: palette.grey[300],
    },
    marginLeft: 0,
    width: "100%",
    [breakpoints.up("sm")]: {
      marginLeft: spacing(1),
      width: "auto",
    },
  },
  searchIcon: {
    width: spacing(9),
    height: "100%",
    position: "absolute",
    pointerEvents: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
  },
  inputRoot: {
    color: "inherit",
    width: "100%",
  },
  inputInput: {
    borderRadius: 4,
    paddingTop: spacing(1),
    paddingRight: spacing(1),
    paddingBottom: spacing(1),
    paddingLeft: spacing(10),
    transition: transitions.create("width"),
    width: "100%",
    [breakpoints.up("sm")]: {
      width: 120,
      "&:focus": {
        width: 200,
      },
    },
  },
});

const HeaderContent = ({ classes, screen }) => (
  <>
    <Typography noWrap color="textSecondary" className={classes.header}>
      Layout v1.1
    </Typography>
    <div className={classes.grow} />
    <div className={classes.search}>
      <div className={classes.searchIcon}>
        <SearchIcon />
      </div>
      <InputBase
        placeholder="Searchâ€¦"
        classes={{
          root: classes.inputRoot,
          input: classes.inputInput,
        }}
      />
    </div>
    {screen === "xs" && (
      <IconButton>
        <MoreVertIcon />
      </IconButton>
    )}
    {isWidthUp("sm", screen) && (
      <>
        <IconButton>
          <NotificationsIcon />
        </IconButton>
        <IconButton>
          <PowerSettingsNewIcon />
        </IconButton>
      </>
    )}
  </>
);

HeaderContent.propTypes = {
  screen: PropTypes.string,
  classes: PropTypes.shape({}).isRequired, // likely populated by withStyles HOC
};
HeaderContent.defaultProps = {
  screen: null,
};

export default withStyles(styles)(HeaderContent);
