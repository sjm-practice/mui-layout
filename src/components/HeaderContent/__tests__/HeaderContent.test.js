/* eslint-env jest */

import React from "react";
import { render } from "@testing-library/react";

import HeaderContent from "../index";

describe("<HeaderContent />", () => {
  it("Should match Snapshot - screen 'xs'", () => {
    const { asFragment } = render(<HeaderContent screen="xs" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("Should match Snapshot - screen 'sm'", () => {
    const { asFragment } = render(<HeaderContent screen="sm" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("Should match Snapshot - screen 'md'", () => {
    const { asFragment } = render(<HeaderContent screen="md" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
