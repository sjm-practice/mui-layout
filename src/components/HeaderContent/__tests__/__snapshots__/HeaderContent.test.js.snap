// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderContent /> Should match Snapshot - screen 'md' 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root HeaderContent-header-171 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-noWrap"
  >
    Layout v1.1
  </p>
  <div
    class="HeaderContent-grow-172"
  />
  <div
    class="HeaderContent-search-173"
  >
    <div
      class="HeaderContent-searchIcon-174"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        search
      </span>
    </div>
    <div
      class="MuiInputBase-root HeaderContent-inputRoot-175"
    >
      <input
        class="MuiInputBase-input HeaderContent-inputInput-176"
        placeholder="Search…"
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    class="MuiButtonBase-root MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        favorite
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        phone
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        camera
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`<HeaderContent /> Should match Snapshot - screen 'sm' 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root HeaderContent-header-86 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-noWrap"
  >
    Layout v1.1
  </p>
  <div
    class="HeaderContent-grow-87"
  />
  <div
    class="HeaderContent-search-88"
  >
    <div
      class="HeaderContent-searchIcon-89"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        search
      </span>
    </div>
    <div
      class="MuiInputBase-root HeaderContent-inputRoot-90"
    >
      <input
        class="MuiInputBase-input HeaderContent-inputInput-91"
        placeholder="Search…"
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    class="MuiButtonBase-root MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        favorite
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        more_vert
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`<HeaderContent /> Should match Snapshot - screen 'xs' 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root HeaderContent-header-1 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-noWrap"
  >
    Layout v1.1
  </p>
  <div
    class="HeaderContent-grow-2"
  />
  <div
    class="HeaderContent-search-3"
  >
    <div
      class="HeaderContent-searchIcon-4"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        search
      </span>
    </div>
    <div
      class="MuiInputBase-root HeaderContent-inputRoot-5"
    >
      <input
        class="MuiInputBase-input HeaderContent-inputInput-6"
        placeholder="Search…"
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    class="MuiButtonBase-root MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        more_vert
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
